kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ExecWebSocketListener.java

```diff
@@ -218,7 +246,9 @@ public void onClosed(WebSocket webSocket, int code, String reason) {
         webSocketClosed.set(true);
         LOGGER.debug("Exec Web Socket: On Close with code:[{}], due to: [{}]", code, reason);
         try {
-            closeExecutor();
+            if (cleanupOnClose.get()) {
```

Comment:
I'm not sure I see the issue with closing the write end of the stream returned by watch.getOutput(). In my mind, once the websocket is closed, we're not going to get any more data to write to the stream, and so it should be safe to close the write end at that point (the rest of the ExecWatch can remain open). InputStreamPumper should not throw an exception in that case, since we can pull remaining data out of the buffer and should hit `-1` on read once all data has been obtained. As it stands, the last read will block, since the write end stays open after the websocket is closed. 

This was kind of the sticking point in my use case, as it's not clear that the InputStreamPumper thread will never finish on its own. By closing the write end of the output and error streams, we could signal to InputStreamPumper that we're done writing, and once all input in the buffer has been read, it's safe to close.